### PR TITLE
Fork5-rebasing

### PR DESCRIPTION
**Pull request by josojo**
It's a rebase to the newest fork.5 contracts of the polygon contracts. I will keep it open until the polygon team merges it to main

### CODE REVIEW

**Comment by edmundedgar** [comment]
This is their code and you're just merging it in, right? Should I review the specific changes?

File: upgrade/timeLockUpgrade.js [hunk 1]
```diff
@@ -54,22 +55,24 @@ async function main() {
...
                     unsafeAllow: ['constructor', 'state-variable-immutable'],
                 },
             );
 
             console.log({ newImplPolygonAddress });
-            console.log("you can verify the new impl address with:")
+            console.log('you can verify the new impl address with:');
             console.log(`npx hardhat verify --constructor-args upgrade/arguments.js ${newImplPolygonAddress} --network ${process.env.HARDHAT_NETWORK}\n`);
-            console.log("Copy the following constructor arguments on: upgrade/arguments.js \n", upgrade.constructorArgs)
+            console.log('Copy the following constructor arguments on: upgrade/arguments.js \n', upgrade.constructorArgs);
```
Comment (edmundedgar) [inline upgrade/timeLockUpgrade.js:70]: Wouldn't this print a literal \n when we want a new line?

Comment (josojo) [reply upgrade/timeLockUpgrade.js:70]: In my terminal, it prints a new line.

**Comment by josojo** [comment]
Yes, its a 1to1 merge from their code. The only major change is the difference in function signature of verifyBatches with the different proof handover.

No, nothing specific to review, just whether it makes sense all together.

**Pull request closed without merging**